### PR TITLE
feat(clerk-js): Introduce experimental support for Google One Tap

### PR DESCRIPTION
**Pull request by panteliselef**
## Description
This PR introduces: 
- `< __experimental_GoogleOneTap/>`
- `__experimental_mountGoogleOneTap()`
- `__experimental_unmountGoogleOneTap()`

Proper support for the PSU flow will be added later. This PR will also be backported to v4 (core-1)

### DOM while is mounted

Screenshot 2024-04-15 at 2 18 12 PM

### SignIn/SignUp (without additional steps)

https://github.com/clerk/javascript/assets/19269911/0e8315b8-e5b0-475a-8ad1-7a0144404f88

### SignUp (with additional steps)

https://github.com/clerk/javascript/assets/19269911/d0217d95-2b4e-457c-a1c8-31201c57dad0

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/react/src/components/uiComponents.tsx [hunk 1]
```diff
@@ -274,3 +275,14 @@ export const OrganizationList = withClerk(({ clerk, ...props }: WithClerkProp<Or
     />
   );
 }, 'OrganizationList');
+
+export const __experimental_GoogleOneTap = withClerk(({ clerk, ...props }: WithClerkProp<OneTapProps>) => {
```
Comment (anagstef) [inline packages/react/src/components/uiComponents.tsx:279]: ❓ Are we sure a component name starting with `_` works well in JSX?

File: .changeset/grumpy-maps-hope.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+'@clerk/clerk-js': minor
+'@clerk/nextjs': minor
+'@clerk/clerk-react': minor
+'@clerk/types': minor
+---
+
+Introduce experimental support for Google One Tap
+- React Component `<__experimental_GoogleOneTap/>`
+- JS `clerk.__experimental_mountGoogleOneTap(node,props)`
```
Comment (anagstef) [inline .changeset/grumpy-maps-hope.md:10]: ❓ Is the `__experimental_` prefix the one we settled for?

Comment (anagstef) [reply .changeset/grumpy-maps-hope.md:10]: I thought it was `experimental_`

**Review by anagstef — APPROVED** [review]

File: packages/types/src/clerk.ts [hunk 3]
```diff
@@ -182,6 +182,14 @@ export interface Clerk {
    */
   unmountSignIn: (targetNode: HTMLDivElement) => void;
 
+  /**
+   * Mounts a sign in flow component at the target element.
+   * @param targetNode Target node to mount the SignIn component.
+   * @param signInProps sign in configuration parameters.
+   */
```
Comment (BRKalow) [inline packages/types/src/clerk.ts:189]:
```suggestion
  /**
   * Mounts a google one tap flow component at the target element.
   * @param targetNode Target node to mount the GoogleOneTap component.
   * @param oneTapProps sign in configuration parameters.
   */
```

File: playground/nextjs/pages/index.tsx [hunk 4]
```diff
@@ -18,6 +19,10 @@ const Home: NextPage = () => {
         />
       </Head>
 
+      <SignedOut>
+        <__experimental_GoogleOneTap />
+      </SignedOut>
```
Comment (BRKalow) [inline playground/nextjs/pages/index.tsx:24]: ❓ do we want to put a signed out check inside the component itself?

File: packages/clerk-js/src/ui/components/GoogleOneTap/one-tap-start.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      /**
+       * Currently it is not possible to display an error in the UI.
+       * As a fallback we simply open the SignIn modal for the user to sign in.
+       */
+      clerk.openSignIn();
+    }
+  }
+
+  const environmentClientID = environment.displayConfig.googleOneTapClientId;
+  const shouldLoadGIS = user?.id && environmentClientID;
```
Comment (BRKalow) [inline packages/clerk-js/src/ui/components/GoogleOneTap/one-tap-start.tsx:48]: ❓ Why do we only load GIS if user.id is defined?

Comment (panteliselef) [reply packages/clerk-js/src/ui/components/GoogleOneTap/one-tap-start.tsx:48]: This is not right, I pushed a fix.

File: playground/nextjs/pages/index.tsx (hunk 4 above)
Comment (panteliselef) [reply playground/nextjs/pages/index.tsx:24]: We do have that check inside `one-tap-start.tsx`. I will be dropping this file change tho.

File: .changeset/grumpy-maps-hope.md (hunk 2 above)
Comment (panteliselef) [reply .changeset/grumpy-maps-hope.md:10]: Pretty sure we settled for `__experimental_` same one we had for the most recent project "passkeys"

File: packages/react/src/components/uiComponents.tsx (hunk 1 above)
Comment (panteliselef) [reply packages/react/src/components/uiComponents.tsx:279]: Using it with our playground did not give me trouble

**Review by BRKalow — APPROVED** [review]

**Pull request merged**